### PR TITLE
Update Action component versions  #54;  Try to support Judger state report  #59

### LINKED ISSUE #54: Update Action component versions
You can reference:
https://github.com/OJ-lab/judger/blob/main/.github/workflows/docker-publish-server.yml

Simply raise a PR with updated version like `@v4` should be fine

### PR DESCRIPTION
**Pull request by Zztrans**
FIX https://github.com/OJ-lab/oj-lab-platform/issues/54

### CODE REVIEW

**Comment by Zztrans** [comment]
FIX https://github.com/OJ-lab/oj-lab-platform/issues/54?

**Comment by Zztrans** [comment]
... I also try to solve https://github.com/OJ-lab/oj-lab-platform/issues/59.
Currently I just simplely parse the `verdictJson`, solve verdictPriority, and get summary of memory, time info, make `mainResult` expose to front.

I dont know how to make good use of `JudgeResult` with `TestPoint`.  T_T

**Comment by slhmy** [comment]
The fix on Action looks good.

However, you might misunderstood the 'state' in #59. It's the state of judger, not the state of judge task.
You may stash your change on judge task and reopen a MR for it.
Judge task will be hard to implement since judger don't currently follows this new design. @Zztrans

File: src/service/task.go [hunk 1]
```diff
@@ -49,3 +59,92 @@ func ReportJudgeTaskResult(
...
+		2: model.JudgeVerdictTimeLimitExceeded,
+		3: model.JudgeVerdictMemoryLimitExceeded,
+		4: model.JudgeVerdictSystemError,
+		5: model.JudgeVerdictWrongAnswer,
+		6: model.JudgeVerdictAccepted,
+	}
+
+	for _, test := range tests {
+		var (
+			tmpVerdict = ""
```
Comment (slhmy) [inline src/service/task.go:99]: This parse func is not good，try to deserialize the JSON by a struct。

**Review by slhmy — CHANGES_REQUESTED** [review]

**Comment by Zztrans** [comment]
ok, I will open a new PR just for https://github.com/OJ-lab/oj-lab-platform/issues/54

File: src/service/task.go (hunk 1 above)
Comment (Zztrans) [reply src/service/task.go:99]: ok

**Pull request closed without merging**